### ISSUE TITLE
Bookmark notification

### ISSUE THREAD

**Issue by cherrielam** [pg-irc/pathways-frontend#892]
The current bookmarks notification is barely usable. On iOS, the notification is down at the bottom and very hard to be seen. On iPhone X + models, the bookmark notification is blocked by the home bar. On Android, the notifications won't even show up. 

We propose a UI change for the bookmark notification so that it is more noticeable and informative for the users. The bookmark is now a dark grey [snackbar](https://reactnativeexample.com/react-native-smart-tip-including-toast-modal-snackbar/) that pops up _vertically above_ the bottom nav. (Reference: https://github.com/PeaceGeeksSociety/pathways-ux-ui/issues/77)

### Mocks
- Zeplin: [See individual screens](https://app.zeplin.io/project/5bc7bfa42e72260acb03aa86/dashboard?seid=5dcb5f0e101ee1bf67e7c315)
- Sketch Cloud: Clickable prototypes - for flow: 
  - [Topic saved/removed](https://sketch.cloud/s/LMGO5/a/zVGyjl/play) 
  - [Service saved/removed](https://sketch.cloud/s/LMGO5/a/JxDAGM/play)

### Acceptance Criteria 
- [ ]  

### Steps for QA: 
1.

**Comment by jparlina** [comment]
Tested on Staging V1.17.0, Server V1.3.23 on:

**1. iPhone 7s (iOS 13.3)**: 
Confirmed with @cherrielam that the wording stays as "Bookmark added/removed" instead of "Topic saved/removed" and "Service saved/removed" since "Bookmark added/removed" has previously been translated to all languages. 
**ISSUES** (per discussion with @cherrielam and @aron-chen-pg):
- Bookmark is too high (see screenshots) - should be just above the navigation bar just as shown on the sketches. 
- Bookmark stayed too long on the screen (see recorded video [here](https://drive.google.com/file/d/1JV6Z1uFrkBe6yrqEwnXSlHV375nH_Yq5/view?usp=sharing)) 

![IMG_3937](https://user-images.githubusercontent.com/48343061/73209846-ed475500-40fd-11ea-8995-ce0c155bef6e.PNG)


![IMG_3938](https://user-images.githubusercontent.com/48343061/73209847-ed475500-40fd-11ea-9aa2-7e253c723218.PNG)


![IMG_3939](https://user-images.githubusercontent.com/48343061/73209848-ed475500-40fd-11ea-99d0-38c27f3af88b.PNG)


![IMG_3940](https://user-images.githubusercontent.com/48343061/73209849-ed475500-40fd-11ea-9330-92f3a3594c87.PNG)


**2. Android V6.0.1 **: 
The notifications look better on Android, consistent with the sketches. 
**ISSUE**: Bookmark notification stays too long on the screen (same as on iOS)

**Comment by aron-chen-pg** [comment]
Tested on Staging V1.17.0, Server V1.3.23 on:

iPhone X:
Bookmark notification's style and position look largely consistent with the mockups.

**Comment by tomy-pg** [comment]
@bensinjin -

By default, Flatlist should not scroll to the top when the component re-renders. However, when a service is bookmarked or unbookmarked, it seems to do this. I think the third party libraries we are using is causing this to happen as this behaviour is not seen when bookmarking topic services. 

Search bookmarking: 

I tried researching what JSX element `InstantSearch` uses under the hood to see if there is anything obvious that is messing up the Flatlist in `InfiniteHitsComponent`. Apparently it uses Context.Provider. https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/react/#removed-support-for-root-in-instantsearch-and-index

I attempted to wrap `InfiniteHitsConnectedComponent` in a ScrollView to see if I can prevent the Flatlist from scrolling. Up until the 10th item in the Flatlist, the Flatlist did not scroll to the top of the when a service was bookmarked. However, after the 10th item, after every bookmark, the Flatlist will scroll to the item in the 10th position. 

Things observed: 
When a service is bookmarked, refine() inside `Search_Input_Component` is being called. Refine is the function used to load results from Algolia.
